library/bignum_mod.h

```diff
@@ -98,10 +98,11 @@ typedef enum {
     /* Skip 1 as it is slightly easier to accidentally pass to functions. */
     /** Montgomery representation. */
     MBEDTLS_MPI_MOD_REP_MONTGOMERY = 2,
-    /** TODO: document this.
-     *
-     * Residues are in canonical representation.
-     */
+    /* Optimised reduction available. This indicates a coordinate modulus (P)
+     * and one of the following available:
+     * - MBEDTLS_ECP_NIST_OPTIM
```

Comment:
For this to fit with the other options, perhaps this should be:

Nist Curve + MBEDTLS_ECP_NIST_OPTIM defined

Reply:
To my unerstanding the `MBEDTLS_ECP_NIST_OPTIM` does not apply to Koblitz curves, curve25519 and curve448. Which is why the wording is as such. 

It is possible to have `MBEDTLS_ECP_NIST_OPTIM` unset and still using a coordinate modulus.

Reply:
I think Paul means that `MBEDTLS_ECP_NIST_OPTIM` is needed but not sufficient to enable a NIST reduction. The specific reduction define is also needed.

Reply:
I have consulted with @paul-elliott-arm  and reworded the comments appropriatly